### PR TITLE
main.cpp passes

### PR DESCRIPTION
**Pull request by maro243**

### CODE REVIEW

File: course02/homework00/skeleton/vector.h [hunk 1]
```diff
@@ -2,7 +2,16 @@
 
 class VectorInt
 {
-	// TODO
+public:
+	int capacity() const;
+	int size() const;
+	int back() const;
```
Comment (david-grs) [inline course02/homework00/skeleton/vector.h]: You should add the corresponding `int& back()` method

File: course02/homework00/skeleton/vector.h [hunk 2]
```diff
@@ -2,7 +2,16 @@
...
+	void push_back(const int&) ;
```
Comment (david-grs) [inline course02/homework00/skeleton/vector.h]: fundamental types should not be passed as const ref

File: course02/homework00/skeleton/vector.h [hunk 3]
```diff
@@ -2,7 +2,16 @@
...
+	void clear();	
+private:
+	static const int mMaxSize = 256;
+	int mData[mMaxSize];
```
Comment (david-grs) [inline course02/homework00/skeleton/vector.h]: Don't use C style array

File: course02/homework00/skeleton/vector.cc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+int VectorInt::size() const
+{
+	return mSize;
+}
+
+int VectorInt::back() const
+{
+	return mData[mSize -1];
```
Comment (david-grs) [inline course02/homework00/skeleton/vector.cc]: At the very least you should add an assert here

File: course02/homework00/skeleton/vector.cc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+void VectorInt::push_back(const int& x)
+{
+	mData[mSize++] = x;
```
Comment (david-grs) [inline course02/homework00/skeleton/vector.cc]: same remark here on assert

**Review by david-grs — APPROVED** [review]
Please make the changes and merge! Thanks

File: course02/homework00/skeleton/vector.cc (hunk 5 above)
Comment (maro243) [reply course02/homework00/skeleton/vector.cc]: not sure what you mean here,  getting the back of an empty vector should be caught, but you can always push_back

Comment (maro243) [reply course02/homework00/skeleton/vector.cc]: oh i guess you mean that i'm not hitting capacity

**Comment by maro243** [comment]
re-opening with new material

**Pull request closed without merging**